src/make_watertight/Cleanup.cpp

```diff
@@ -0,0 +1,390 @@
+    if (moab::MB_SUCCESS != result) {
+      std::cout << "result=" << result << std::endl;
+    }
+    assert(moab::MB_SUCCESS == result);
+
+    // get the internal edges
+    moab::Range internal_edges = subtract(edges, skin_edges);
+
+    for (moab::Range::iterator j = internal_edges.begin(); j != internal_edges.end(); j++) {
+      int n_internal_edges = internal_edges.size();
+      std::cout << "edge=" << *j << std::endl;
+      MBI()->list_entity(*j);
+      assert(moab::MB_SUCCESS == result);
+
+      // get endpts
+      const moab::EntityHandle* endpts;
+      int n_verts;
+      result = MBI()->get_connectivity(*j, endpts, n_verts);
+      assert(moab::MB_SUCCESS == result);
+      assert(2 == n_verts);
+
+      // does another edge exist w the same endpts? Why would it?
+      moab::Range duplicate_edges;
+      result = MBI()->get_adjacencies(endpts, 2, 1, true, duplicate_edges);
+      assert(moab::MB_SUCCESS == result);
+      if (1 < duplicate_edges.size())
+        MBI()->list_entities(duplicate_edges);
+      assert(1 == duplicate_edges.size());
+
+      // if the edge length is less than MERGE_TOL do nothing
+      if (FACET_TOL < gen->dist_between_verts(endpts[0], endpts[1]))
+        continue;
+
+      // quick check
+      for (moab::Range::iterator k = internal_edges.begin(); k != internal_edges.end(); k++) {
+        const moab::EntityHandle* epts;
+        int n_vts;
+        result = MBI()->get_connectivity(*k, epts, n_vts);
+        assert(moab::MB_SUCCESS == result);
+        assert(2 == n_vts);
+        // The skin edges/verts cannot be moved, therefore both endpoints cannot
+        // be on the skin. If they are, continue.
+        moab::Range adj_edges0;
+        result = MBI()->get_adjacencies(&epts[0], 1, 1, true, adj_edges0);
+        assert(moab::MB_SUCCESS == result);
+        if (3 > adj_edges0.size()) {
+          std::cout << "adj_edges0.size()=" << adj_edges0.size()
+                    << " epts[0]=" << epts[0] << std::endl;
+          MBI()->list_entity(epts[0]);
+          assert(moab::MB_SUCCESS == result);
+        }
+        assert(3 <= adj_edges0.size());
+        moab::Range adj_skin_edges0 = intersect(adj_edges0, skin_edges);
+        bool endpt0_is_skin;
+        if (adj_skin_edges0.empty())
+          endpt0_is_skin = false;
+        else
+          endpt0_is_skin = true;
+
+        moab::Range adj_edges1;
+        result = MBI()->get_adjacencies(&epts[1], 1, 1, true, adj_edges1);
+        assert(moab::MB_SUCCESS == result);
+        if (3 > adj_edges1.size()) {
+          std::cout << "adj_edges1.size()=" << adj_edges1.size()
+                    << " epts[1]=" << epts[1] << std::endl;
+          MBI()->list_entity(epts[1]);
+          assert(moab::MB_SUCCESS == result);
+        }
+        assert(3 <= adj_edges1.size());
+      }
+
+
+      // The skin edges/verts cannot be moved, therefore both endpoints cannot
+      // be on the skin. If they are, continue.
+      moab::Range adj_edges0;
```

Comment:
Reducing Future Bugs: The check for `endpt0_is_skin` and `endpt1_is_skin` below look identical.  To save on code and potential future bugs, it may be worth it to put these two values to check into an array and turn this into a `for` loop.

Reply:
It looks to me like the main problem here is that the logic is overly verbose and the variables are far set from where they are used in one case. As the `Range::empty()` call already returns a Boolean, I've moved initialization of the `endpt0_is_skin` and `endpt1_is_skin` closer to where they're used and set them to `!adj_skin_edges0/1.empty()` respectively. The logic is now small enough to simply put into the if statement, but I felt the Booleans added enough clarity to keep them.

With those updates, I don't think the array for loop approach is necessary anymore.